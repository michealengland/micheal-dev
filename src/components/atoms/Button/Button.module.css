.button {
  @apply bg-primary-purple2;
}
