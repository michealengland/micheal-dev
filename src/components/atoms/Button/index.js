import React from 'react'
import Link from '@components/atoms/link'
import ProptTypes from 'prop-types'
import styles from './Button.module.css'
import cn from 'classnames'

const Button = ({className, label, to}) => (
  <Link className={cn(styles.button, className)} to={to}>
    {label}
  </Link>
)

export default Button

Button.propTypes = {
  className: ProptTypes.string,
  label: ProptTypes.string,
  to: ProptTypes.string,
}
