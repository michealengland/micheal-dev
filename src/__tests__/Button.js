import React, {Fragment} from 'react'
import renderer from 'react-test-renderer'
import Button from '@components/atoms/Button'

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Fragment>
          <Button
            label="Internal Link"
            to="/"
          />
          <Button
            label="External Link"
            to="https://micheal.dev"
          />
          <Button
            to="https://micheal.dev"
          />
        </Fragment>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
