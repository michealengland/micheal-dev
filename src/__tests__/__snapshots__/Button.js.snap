// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
Array [
  <a
    className="button"
    href="/"
  >
    Internal Link
  </a>,
  <a
    className="button"
    href="https://micheal.dev"
  >
    External Link
  </a>,
  <a
    className="button"
    href="https://micheal.dev"
  >
    https://micheal.dev
  </a>,
]
`;
